chingu-voyages/v50-tier2-team-14
# Feature/Payment system

# Description

- Created a mock payment system where a user clicks on a coin to add mock credits to their account. when clicking on 'place order button' available credits are compared to total order and a success (green) or error (red) message displays. When green message is displayed, the cart is reset and available credits are updated. Red message appears in two cases: 

1. When the cart is empty => message says your cart is empty
2. When there are not enough credits => message says not enough credits
3. 
 - Also added functionality to store cart data in local storage. Now when a user refreshes the page, cart content is not reset.


## Main changes explained:
- Created Payment system component
- Created error alert component
- Added replaceCart action to cart-slice in Redux
…

## How to test:
1. check into current branch
4. do `npm install` and `npm start` to run this PR locally
5. go to localhost:3000
6. add items from menu to cart 
7. refresh the page, verify that the cart remains the same
8. proceed to checkout and click on place order
9. verify that you see green or red messages depending on situation
10. add credits to your account by clicking on the coin image and verify that it works and relevant success/error messages are displayed

## Screenshots or videos of changes:

Desktop look:
![Screenshot 2024-08-08 at 12 16 16 PM](https://github.com/user-attachments/assets/1450d511-2a0c-491d-bff1-21c750c6d0cb)


## Note:
IMPORTANT: there is a bug which I haven't been able to figure out just yet. Sometimes when an order is placed and the cart is not empty, you can see an error message saying that the cart is empty. After clicking on the button second time, the order is placed successfully. Also sometimes after placing an order the updated credit displayed above the coin image shows a random number. I have not been able to figure out how to solve these two issues. If you have any ideas, please let me know.
